internal/api/mail.go

```diff
@@ -650,21 +650,22 @@ func (a *API) sendEmail(r *http.Request, tx *storage.Connection, u *models.User,
 	}
 
 	if config.Hook.SendEmail.Enabled {
-		// When secure email change is disabled, we place the token for the new email on emailData.Token
-		if emailActionType == mail.EmailChangeVerification && !config.Mailer.SecureEmailChangeEnabled && u.GetEmail() != "" {
-			otp = otpNew
-		}
-
 		emailData := mail.EmailData{
 			Token:           otp,
 			EmailActionType: emailActionType,
 			RedirectTo:      referrerURL,
 			SiteURL:         externalURL.String(),
 			TokenHash:       tokenHashWithPrefix,
 		}
-		if emailActionType == mail.EmailChangeVerification && config.Mailer.SecureEmailChangeEnabled && u.GetEmail() != "" {
-			emailData.TokenNew = otpNew
-			emailData.TokenHashNew = u.EmailChangeTokenCurrent
+		if emailActionType == mail.EmailChangeVerification {
+			// When secure email change is disabled, we place the token for the new email on emailData.Token
+			if !config.Mailer.SecureEmailChangeEnabled {
+				// Token Hash should already be set above
+				emailData.Token = otpNew
```

Comment:
does this mean that the actual value in `TokenHash` will be different from the `otpNew` ?